### PR TITLE
SNOW-1269024 Installing multiple Java versions for test purposes

### PR DESCRIPTION
**Pull request by sfc-gh-krosinski**
# Overview

SNOW-1269024 - adding verification on different versions of JRE
Building and unit testing process remains the same (Java8) but e2e testing is run on three different versions of JAVA: 8, 11 and 17. This means that there are 3 times more testing jobs, but they all run in parallel, hence total run time should remain ~the same. 

## Pre-review checklist
- [ ] This change should be part of a Behavior Change Release. See [go/behavior-change](http://go/behavior-change).
- [x] This change has passed Merge gate tests
- [ ] Snowpipe Changes
- [ ] Snowpipe Streaming Changes
- [x] This change is TEST-ONLY
- [ ] This change is README/Javadocs only
- [ ] This change is protected by a config parameter <PARAMETER_NAME> eg `snowflake.ingestion.method`.
    - [ ] `Yes` - Added end to end and Unit Tests. 
    - [x] `No` - This is test only change
- [ ] Is his change protected by parameter <PARAMETER_NAME> on the server side?
    - [ ] The parameter/feature is not yet active in production (partial rollout or PrPr, see [Changes for Unreleased Features and Fixes](http://go/ppp-prpr)).
    - [ ] If there is an issue, it can be safely mitigated by turning the parameter off. This is also verified by a test (See [go/ppp](http://go/ppp)).

*normal* 

Longer test runs

### CODE REVIEW

**Review by sfc-gh-tzhang — APPROVED** [review]
LGTM, please see if @sfc-gh-japatel could take a look

File: .github/workflows/End2EndTestApache.yml [hunk 1]
```diff
@@ -18,9 +18,12 @@ jobs:
     - name: Checkout Code
       uses: actions/checkout@v2
     - name: "Install Java 8"
```
[the remark this answers is not in the document]
Comment (sfc-gh-krosinski) [reply .github/workflows/End2EndTestApache.yml:21]: Decided to go with different approach - here only Java 8 is installed, hence the name remains the same.

**Review by sfc-gh-japatel — COMMENTED** [review]
any plans for confluent end to end test?

File: .github/workflows/End2EndTestApache.yml [hunk 2]
```diff
@@ -14,13 +14,15 @@ jobs:
       matrix:
         apache_kafka_version: [ '2.5.1', '2.8.1', '3.2.1' ]
         snowflake_cloud: [ 'AWS', 'AZURE', 'GCS' ]
+        java_test_version: [ '8', '11', '17' ]
     steps:
     - name: Checkout Code
       uses: actions/checkout@v2
     - name: "Install Java 8"
```
Comment (sfc-gh-japatel) [inline .github/workflows/End2EndTestApache.yml:21]: why do we need this step? as in we are installing java 8 here but using different versions below.

Comment (sfc-gh-krosinski) [reply .github/workflows/End2EndTestApache.yml:21]: the whole point of my change is to use two different versions of java - java 8 for building and unit testing but then 8, 11 and 17 as jre to run the app and verify it against given runtime environment

**Review by sfc-gh-wtrefon — APPROVED** [review]
LGTM

**Review by sfc-gh-lshcharbaty — APPROVED** [review]

**Review by sfc-gh-dseweryn — APPROVED** [review]
Awesome 💪

**Pull request merged**